WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/Preview/ProductDetailPreviewViewModel.swift

```diff
@@ -187,22 +253,81 @@ final class ProductDetailPreviewViewModel: ObservableObject {
     }
 }
 
+// MARK: - Undo edits
+//
+extension ProductDetailPreviewViewModel {
+    func undoEdits(in field: ProductDetailField) {
+        guard let generatedAIProduct else {
+            return
+        }
+
+        guard let option = options[safe: selectedOptionIndex] else {
+            return
+        }
+
+        switch field {
+        case .name:
+            options[selectedOptionIndex] = NameSummaryDescOption(name: generatedAIProduct.names[selectedOptionIndex],
+                                                                 shortDescription: option.shortDescription,
+                                                                 description: option.description)
+        case .shortDescription:
+
+            options[selectedOptionIndex] = NameSummaryDescOption(name: option.name,
+                                                                 shortDescription: generatedAIProduct.shortDescriptions[selectedOptionIndex],
+                                                                 description: option.description)
+        case .description:
+            options[selectedOptionIndex] = NameSummaryDescOption(name: option.name,
+                                                                 shortDescription: option.shortDescription,
+                                                                 description: generatedAIProduct.descriptions[selectedOptionIndex])
+        }
```

Comment:
What do you think about avoiding the duplication:

```suggestion
        let name = updatedField != .name ? productName :
            generatedAIProduct.names[selectedOptionIndex]

        let shortDescription = updatedField != .shortDescription ? productShortDescription :
            generatedAIProduct.shortDescriptions[selectedOptionIndex]

        let description = updatedField != .description ? productDescription :
            generatedAIProduct.descriptions[selectedOptionIndex]

        options[selectedOptionIndex] = NameSummaryDescOption(name: name, shortDescription: shortDescription, description: description)
```